### DISCUSSION TITLE
Support multiple languages

### DISCUSSION THREAD

**Discussion by eikowagenknecht** [eikowagenknecht/lootscraper#456] [Feature Requests]
Most platforms display the offerings in the language the browser sends. Currently it is hardcoded to "en-US", but it would be nice to (additionally) add more languages. The question is how those languages should be handled. Add all translations in the RSS feeds for the entry? Create seperate feeds for every language?

**Comment by louanfontenele** [comment]
It would be great if the bot also had an option for multi-languages.

I can translate to Brazilian Portuguese if you want.

**Comment by louanfontenele** [comment]
@eikowagenknecht any news about multiple languages? :D

**Comment by eikowagenknecht** [comment]
@louanfontenele I think this will take quite a while, so unfortunately no, neither news nor an estimated date ;-)

**Comment by azurejoga** [comment]
Hi, any results from this? I would love to help with implementation for new languages

**Comment by eikowagenknecht** [comment]
Hey! Some exciting news first:

I'm currently rewriting LootScraper in Node.js/TypeScript. This transition from Python comes after gaining more experience with the Node.js ecosystem and encountering some performance and stability challenges with the current version.
The rewrite is nearly feature-complete and has amazing , and I expect to finish within the next few weeks. After that, I'm planning to implement new features, with internationalization being a priority.

**Proposed i18n Approach:**
The simplest implementation would be a global language setting in the config file, meaning:

- All users would share the same language setting
- Only static text would need translation
- Scraping would run in the configured language

**Feedback Needed**
I'd appreciate your input on:

- Would a single global language setting meet your needs?
- Are you planning to use this for self-hosted instances or the public Telegram bot/RSS feeds?

Note: Supporting multiple languages for the public services would significantly increase complexity, particularly around the scraping process, because multiple languages would need to be scraped, stored etc.

**Reply by louanfontenele** [reply]
- Yes, only english can be provided, and I can translate to Brazilian Portuguese
- Should be nice the entire project can be translated, so for example I use your Telegram bot, I can choose the language of posts, so I will recieve the post on my own language, the same will be aplied at homepage, and other bots.
